BLSQ/iaso
# IA-3446 Org Unit Type restrictions (frontend)

In our use case, "admin" users should be able to edit/add new OUs at level 6 for instance (SSC, CAC) but should clearly not be allowed to modify levels 1 to 5 (even within their geographic permission).

Related JIRA tickets : IA-3446

## Self proofreading checklist


## Changes

- Added `editable_org_unit_types` to the profile API and frontend.
- Restricted selection to valid org unit types when editing users and setting `editable_org_unit_types`.
- Allowed changes only to writable org unit types when editing individual org units.
- Disabled multi-selection for bulk editing to ensure only writable org unit types can be selected.
- Limited default image edits to writable org unit types.
- Restricted linking of instances to org units based on writable types (requires `SHOW_LINK_INSTANCE_REFERENCE` feature flag, and the instance must be of the org unit's reference form).

## Testing Steps

1. Ensure you have a proper org unit type hierarchy set up in your dataset.
2. Create a new user and assign them write access for org units, along with read and write permissions for forms and submissions.
3. While editing the user, go to the last tab and select some org unit types as editable.
4. Log in with the new user account:
    - Search for all org units and verify that only the selected org unit types are available for bulk updates.
    - Open the bulk update dialog and attempt to change org unit types; you should only be able to select the editable types.
    - Close the dialog.
5. Edit an org unit of an editable type:
    - Confirm that you can modify all details, including the default image (only editable types should be available in the list).
6. Edit an org unit of a non-editable type:
    - Verify that you cannot change any org unit information or update the default image.
7. Find an instance of a reference form for an org unit:
    - Open the instance's details and ensure that the option to link the instance to the org unit is available only if the org unit type is editable.

## Print screen / video


Uploading Screen Recording 2024-10-08 at 14.18.09.mov…



## Notes

- This is going along with this [PR](https://github.com/BLSQ/bluesquare-components/pull/166)
- Should we limit access to org unit types edition/creation for an user that can only create certain org unit types
- Should we filter out non editable types from change config creation/edition
- Stil need to write cypress tests to cover those cases
